### PR TITLE
Domain updates

### PR DESCRIPTION
**Pull request by BryanRumsey**
## Domain article Tabs
- [x] Geometries
- [x] Lattices
- [x] Transformations
- [x] Actions

### CODE REVIEW

**Comment by mdip226** [comment]
Notes from my build:
pip:
```
statsmodels 0.13.5 requires packaging>=21.3, but you'll have packaging 20.4 which is incompatible.
numba 0.56.4 requires numpy<1.24,>=1.18, but you'll have numpy 1.24.1 which is incompatible.
```
```
npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
npm WARN deprecated popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokidar 3 with 15x fewer dependencies
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
```
```
bash -c "if [ ! -d \"/home/mdip/bs/stochss/local_data\" ] && [ -z ]; then mkdir /home/mdip/bs/stochss/local_data; mkdir /home/mdip/bs/stochss/local_data/Examples;cp -r public_models/* /home/mdip/bs/stochss/local_data/Examples;fi"
python3 launch_webbrowser.py &
docker run --rm \
	--name stochss-lab \
	--env-file .env \
	-v /home/mdip/bs/stochss/local_data:/home/jovyan/ \
	-p 8888:8888 \
	stochss-lab:latest \
	bash -c "cd /home/jovyan; start-notebook.sh "
Welcome to the StochSS webbrowser launcher!
Initializing docker...
Traceback (most recent call last):
  File "launch_webbrowser.py", line 25, in <module>
    docker_client=docker.from_env()
AttributeError: module 'docker' has no attribute 'from_env'
```

**Review by briandrawert — APPROVED** [review]

**Review by seanebum — APPROVED** [review]

**Comment by briandrawert** [comment]
In the Shapes section, add a bullet point:
"Fillable must be check for a shape to be used in a Fill Action (below). Non-Fillable shapes can be used to set properties, remove particles, or in a Combinatory Geometry"

File: client/domain-view/templates/typesView.pug [hunk 1]
```diff
@@ -42,14 +46,18 @@ div#domain-types-editor.card
...
 
             h6.inline <b>Number of Particles</b>
 
+          div.col-sm-2
+
+            h6.inline <b>Remove</b>
+
         div.my-3(data-hook="edit-domain-types-list")
 
         button.btn.btn-outline-primary.box-shadow(data-hook="add-domain-type") Add Type
```
Comment (mdip226) [inline client/domain-view/templates/typesView.pug:63]: line 77: should be "Number" of Particles

**Review by mdip226 — CHANGES_REQUESTED** [review]
line 77:
./client/domain-view/templates/typesView.pug:            h6.inline <b>Numbe of Particles</b>

**Comment by mdip226** [comment]
This happened when I didn't assign a type to the particles in a fill action
```
[W 12:50:28.730 NotebookApp] 406 POST /stochss/api/spatial-model/domain-plot (172.17.0.1) 4.07ms referer=http://localhost:8888/stochss/domain/edit?path=test.proj/model.wkgp/model.smdl
2023-02-15 12:54:36,265 - GillesPy2 - ERROR - Job errors: Particles must be assigned a type_id.
Traceback (most recent call last):
  File "/stochss/stochss/handlers/util/scripts/start_job.py", line 118, in <module>
    job.run(verbose=args.verbose)
  File "/stochss/stochss/handlers/util/spatial_simulation.py", line 122, in run
    results = self.s_py_model.run(**kwargs)
  File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1055, in run
    return sol.run(number_of_trajectories=number_of_trajectories, seed=seed, timeout=timeout,
  File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 547, in run
    self.compile(debug=debug, profile=profile)
  File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 455, in compile
    stoich_matrix, dep_graph = self.model.compile_prep()
  File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1000, in compile_prep
    self.domain.compile_prep(allow_all_types=allow_all_types)
  File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/domain.py", line 490, in compile_prep
    raise DomainError("Particles must be assigned a type_id.")
spatialpy.core.spatialpyerror.DomainError: Particles must be assigned a type_id.

[E 12:54:36 StochSS start_job.py:70] Job errors: Particles must be assigned a type_id.
    Traceback (most recent call last):
      File "/stochss/stochss/handlers/util/scripts/start_job.py", line 118, in <module>
        job.run(verbose=args.verbose)
      File "/stochss/stochss/handlers/util/spatial_simulation.py", line 122, in run
        results = self.s_py_model.run(**kwargs)
      File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1055, in run
        return sol.run(number_of_trajectories=number_of_trajectories, seed=seed, timeout=timeout,
      File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 547, in run
        self.compile(debug=debug, profile=profile)
      File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 455, in compile
        stoich_matrix, dep_graph = self.model.compile_prep()
      File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1000, in compile_prep
        self.domain.compile_prep(allow_all_types=allow_all_types)
      File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/domain.py", line 490, in compile_prep
        raise DomainError("Particles must be assigned a type_id.")
    spatialpy.core.spatialpyerror.DomainError: Particles must be assigned a type_id.
    
[<StreamHandler <stderr> (NOTSET)>]

```

**Comment by BryanRumsey** [comment]
> In the Shapes section, add a bullet point: "Fillable must be check for a shape to be used in a Fill Action (below). Non-Fillable shapes can be used to set properties, remove particles, or in a Combinatory Geometry"

Done

**Comment by BryanRumsey** [comment]
> This happened when I didn't assign a type to the particles in a fill action
> 
> ```
> [W 12:50:28.730 NotebookApp] 406 POST /stochss/api/spatial-model/domain-plot (172.17.0.1) 4.07ms referer=http://localhost:8888/stochss/domain/edit?path=test.proj/model.wkgp/model.smdl
> 2023-02-15 12:54:36,265 - GillesPy2 - ERROR - Job errors: Particles must be assigned a type_id.
> Traceback (most recent call last):
>   File "/stochss/stochss/handlers/util/scripts/start_job.py", line 118, in <module>
>     job.run(verbose=args.verbose)
>   File "/stochss/stochss/handlers/util/spatial_simulation.py", line 122, in run
>     results = self.s_py_model.run(**kwargs)
>   File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1055, in run
>     return sol.run(number_of_trajectories=number_of_trajectories, seed=seed, timeout=timeout,
>   File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 547, in run
>     self.compile(debug=debug, profile=profile)
>   File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 455, in compile
>     stoich_matrix, dep_graph = self.model.compile_prep()
>   File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1000, in compile_prep
>     self.domain.compile_prep(allow_all_types=allow_all_types)
>   File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/domain.py", line 490, in compile_prep
>     raise DomainError("Particles must be assigned a type_id.")
> spatialpy.core.spatialpyerror.DomainError: Particles must be assigned a type_id.
> 
> [E 12:54:36 StochSS start_job.py:70] Job errors: Particles must be assigned a type_id.
>     Traceback (most recent call last):
>       File "/stochss/stochss/handlers/util/scripts/start_job.py", line 118, in <module>
>         job.run(verbose=args.verbose)
>       File "/stochss/stochss/handlers/util/spatial_simulation.py", line 122, in run
>         results = self.s_py_model.run(**kwargs)
>       File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1055, in run
>         return sol.run(number_of_trajectories=number_of_trajectories, seed=seed, timeout=timeout,
>       File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 547, in run
>         self.compile(debug=debug, profile=profile)
>       File "/opt/conda/lib/python3.8/site-packages/spatialpy/solvers/solver.py", line 455, in compile
>         stoich_matrix, dep_graph = self.model.compile_prep()
>       File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/model.py", line 1000, in compile_prep
>         self.domain.compile_prep(allow_all_types=allow_all_types)
>       File "/opt/conda/lib/python3.8/site-packages/spatialpy/core/domain.py", line 490, in compile_prep
>         raise DomainError("Particles must be assigned a type_id.")
>     spatialpy.core.spatialpyerror.DomainError: Particles must be assigned a type_id.
>     
> [<StreamHandler <stderr> (NOTSET)>]
> ```

This is an accurate error although it should not launch a run request if this error occurs.  This will be addressed in #1429.  I fixed the error display in the domain editor.

**Review by mdip226 — APPROVED** [review]

**Pull request merged**